### ISSUE TITLE
numpy.ma.masked_invalid() return object triggers `no-member` for `mask` attribute

### ISSUE THREAD

**Issue by arekfu** [PyCQA/pylint#5715]
### Bug description

This bug report is very similar to issue #3342. I slightly modified the reproducer:
```python
"""Quick test to show pylint error."""

import numpy as np

DATA = np.array([np.nan, 2, 3])
DATA = np.ma.masked_invalid(DATA)
print(DATA.mask)
assert isinstance(DATA, np.ma.MaskedArray)
```

### Command used

```shell
pylint --extension-pkg-allow-list=numpy numpy_pylint.py
```

### Pylint output

```shell
************* Module numpy_pylint
numpy_pylint.py:7:6: E1101: Instance of 'ndarray' has no 'mask' member (no-member)

------------------------------------------------------------------
Your code has been rated at 3.75/10 (previous run: 3.75/10, +0.00)
```

### Expected behavior

No warning should be emitted.

### Pylint version

```shell
pylint 2.12.0
astroid 2.9.0
Python 3.9.1 (default, Dec  8 2020, 03:26:34) 
[GCC 5.4.0 20160609]
```

### OS / Environment

Ubuntu 16.04

**Comment by clavedeluna** [comment]
At what point should issues such as this one, with "needs astroid update" / "needs astroid brain" be closed?

- as soon as astroid PR is merged
- ^ + we add a corresponding pylint test
- ^^ + new astroid version is released with the fix

?

**Comment by Pierre-Sassoulas** [comment]
Proposition 2 and 3 are the same one as a pylint test actually testing something would not pass without astroid being upgraded in pylint first. We choose between closing as soon as the MR in astroid is merged on a case by case basis, depending on how much pylint specific code is involved. For brains, I think a test in astroid is enough.
